integration/emr/alluxio-emr.sh

```diff
@@ -10,99 +10,217 @@
+# Arg 4. (Optional) Delimeter for additional properties
+# Arg 5. (Optional) Extra Alluxio Options. These will be appended to
+#        alluxio-site.properties. Multiple options can be specified using Arg 4
+#        as a delimiter (ex. alluxio.user.file.writetype.default=CACHE_THROUGH;alluxio.user.file.readtype.default=CACHE)
+
+set -o errexit  # exit when a command fails - append "|| true" to allow a
+                # command to fail
+set -o nounset  # exit when attempting to use undeclared variables
+
+
+set -x # debugging
+
+# script constants
+ALLUXIO_HOME=/opt/alluxio
+ALLUXIO_SITE_PROPERTIES=${ALLUXIO_HOME}/conf/alluxio-site.properties
+
+# Downloads a file to the local machine from a remote HTTP(S) or S3 URI into the cwd
+#
+# Args:
+#   $1: URI - the remote location to retrieve the file from
+download_file() {
+  local uri=$1
+
+  if [[ "${uri}" == s3://* ]]
+  then
+    aws s3 cp ${uri} ./
+  else
+    # TODO Add metadata header tag to the wget for filtering out in download metrics.
+    wget -nv "${uri}"
+  fi
+}
+
+# Run a command as a specific user
+# Assumes the provided user already exists on the system and user running script has sudo access
+#
+# Args:
+#   $1: user - the username to run the command as
+#   $2: command - the command to run
+doas() {
+  local user=$1
+  local cmd=$2
+  sudo runuser -l ${user} -c "${cmd}"
+}
+
+# Appends a property KV pair to the alluxio-site.properties file
+#
+# Args:
+#   $1: property name
+#   $2: property value
+append_alluxio_property() {
+  local property=$1
+  local value=$2
+
+  # OK to fail in this section
+  set +o errexit
+  # /opt/alluxio/conf must exist for this to work
+  grep -q ${property} ${ALLUXIO_SITE_PROPERTIES} 2> /dev/null
+  local rv=$?
+  set -o errexit # errors not ok anymore
+  echo $rv
+  if [ $rv -ne 0 ]; then
+    echo "append prop"
+    doas alluxio "echo '${property}=${value}' >> ${ALLUXIO_SITE_PROPERTIES}"
+  fi
+}
+
+# Calculates the default memory size as 1/3 of the total system memory
+#
+# Echo's the result to stdout. To store the return value in a variable use
+# val=$(get_defaultmem_size)
+get_default_mem_size() {
+  local mem_div=3
+  local phy_total=$(free -m | grep -oP '\d+' | head -n1)
+  local mem_size=$(( ${phy_total} / ${mem_div} ))
+  echo "${mem_size}MB"
+}
+
+# Gets the region of the current EC2 instance
+get_aws_region() {
+  local ec2_avail_zone=$(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone)
+  local ec2_region="$(echo "${ec2_avail_zone}" | sed 's/[a-z]$//')"
+  echo "${ec2_region}"
+}
+
+main() {
+  local alluxio_tarball
+  local root_ufs_uri
+  local site_properties_uri
+  local property_delimeter
+  local delimited_properties
+  alluxio_tarball=${1}
+  root_ufs_uri=${2}
+  site_properties_uri=${3:-""}
+  property_delimeter=${4:-";"}
+  delimited_properties=${5:-""}
+
+  # Create user
+  sudo groupadd alluxio -g 600 || true
```

Comment:
Do we want this to have the || true appended? If the user isn't successfully created, most following commands will fail

Reply:
This was some debugging code. I'll remove. good catch.